R/calcCapacityFactorHist.R

```diff
@@ -1,168 +1,120 @@
 #' @title calc Capacity Factor
 #' @description provides capacity factor values
 #'
-#' @param subtype data subtype. Either "wind" or "windoff"
+#' @param subtype data subtype. Either "windoff" or none
 #'
 #' @return magpie object of the capacity factor data
 #' @author Renato Rodrigues, Stephen Bi
 #' @examples
 #' \dontrun{
 #' calcOutput("CapacityFactor")
 #' }
-#'
-calcCapacityFactorHist <- function(subtype){
+#' 
+
 
-  if (subtype == "wind") {
+calcCapacityFactorHist <- function(subtype = "none"){
 
   #mapping of remind technology names to IRENA categories
-  rem_Irena_map <- data.frame(rem=c("hydro","wind","spv","csp","bioigcc","geohdr"),
-                              irena= c("Renewable hydropower","Wind","Solar photovoltaic","Concentrated solar power", "Bioenergy","Geothermal"))
+  mapping_Irena <- tribble(
+    ~irena,                      ~remind,
+    "Renewable hydropower",      "hydro",
+    "Onshore wind energy",       "windon",
+    "Offshore wind energy",      "windoff",
+    "Solar photovoltaic",        "spv",
+    "Concentrated solar power",  "csp",
+    "Bioenergy",                 "bioigcc",
+    "Geothermal",                "geohdr"
+  )
+  
+  if (subtype == "windoff") {
+    mapping_Irena$remind[mapping_Irena$remind == "windon"] <- "wind"
+  }
+
   # Read capacity factor inputs
-  hist_cap <- readSource(type="IRENA",subtype="Capacity")/1000 # converting from MW to GW
-  hist_gen <- readSource("IRENA", subtype = "Generation")# Units are GWh
-  # Calculate 2015 capacity factor for relevant technologies
-  cf_realworld <- hist_gen[,2015,rem_Irena_map$irena]/(8760*hist_cap[,2015,rem_Irena_map$irena])
+  hist_cap <- readSource(type = "IRENA", subtype = "Capacity") / 1000 # converting from MW to GW
+  hist_gen <- readSource(type = "IRENA", subtype = "Generation") # Units are GWh
+
+  # Calculate 2015 capacity factor for relevant technologies, using 2*generation(t) / (cap(t) + cap(t-1) )
+  # then average over 5 years for 2015
+  
+  # load 5 years of generation
+  hist_gen2 = hist_gen[,seq(2013,2017,1), mapping_Irena$irena]%>% 
+    as.data.frame() 
+  
+  # load 5 years of capacity
+  hist_cap2a = hist_cap[,seq(2012,2016,1), mapping_Irena$irena] %>% 
+    as.data.frame() %>% 
+    mutate(Year = as.factor(as.integer(as.character(.data$Year)) + 1)) %>% 
+    select(.data$Year, .data$Region, .data$Data1, 'cap1' ='Value')
```

Comment:
this line and the following below of the same type yield warnings
```
Use of .data in tidyselect expressions was deprecated in tidyselect 1.2.0.
ℹ Please use `"Value"` instead of `.data$Value
```

You can solve them easily by using strings as parameters in select, i.e. 
`select("Year", "Region", "Data1", "cap1" = "Value")`